### PR TITLE
wasm middleware: change path property to url

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [n/a] Commands include options for Linux, MacOS, and Windows within codetabs
- [n/a] New file and folder names are globally unique
- [n/a] Page references use shortcodes instead of markdown or URL links
- [n/a] Images use HTML style and have alternative text
- [n/a] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This changes the configuration item for the Wasm binary from path to
url. In doing so, we ease migration from other sources including HTTP
and OCI.

## Issue reference

https://github.com/dapr/components-contrib/pull/2817 - change that results in this update
https://github.com/dapr/components-contrib/issues/2700 - the change enables this issue to complete in 1.12

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
bad link for AWS fixed here https://github.com/dapr/docs/pull/3360

**Comment by codefromthecrypt** [comment]
actually real link fix that needs to go to v1.11 is [3356](https://github.com/dapr/docs/pull/3356), already in v1.10

**Comment by codefromthecrypt** [comment]
rebased

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 1]
```diff
@@ -17,9 +17,10 @@ binary. In other words, you can extend Dapr using external files that are not
 pre-compiled into the `daprd` binary. Dapr embeds [wazero](https://wazero.io)
 to accomplish this without CGO.
 
-Wasm modules are loaded from a filesystem path. On Kubernetes, see [mounting
-volumes to the Dapr sidecar]({{< ref kubernetes-volume-mounts.md >}}) to configure
-a filesystem mount that can contain Wasm modules.
+Wasm modules are loaded from a URL. For example, a URL "file://rewrite.wasm"
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
Wasm binaries are loaded from a URL. For example, the URL `file://rewrite.wasm`
```

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 2]
```diff
@@ -17,9 +17,10 @@ binary. In other words, you can extend Dapr using external files that are not
...
+loads "rewrite.wasm" from the current directory of the process. On Kubernetes,
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
loads `rewrite.wasm` from the current directory of the process. On Kubernetes,
```

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 3]
```diff
@@ -17,9 +17,10 @@ binary. In other words, you can extend Dapr using external files that are not
...
+see [mounting volumes to the Dapr sidecar]({{< ref kubernetes-volume-mounts.md >}})
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
see [How to: Mount Pod volumes to the Dapr sidecar]({{< ref kubernetes-volume-mounts.md >}})
```

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 4]
```diff
@@ -32,18 +33,18 @@ spec:
...
 
 Minimally, a user must specify a Wasm binary implements the [http-handler](https://http-wasm.io/http-handler/).
 How to compile this is described later.
 
-| Field    | Details                                                        | Required | Example        |
-|----------|----------------------------------------------------------------|----------|----------------|
-| path     | A relative or absolute path to the Wasm binary to instantiate. | true     | "./hello.wasm" |
+| Field | Details                                                        | Required | Example        |
+|-------|----------------------------------------------------------------|----------|----------------|
+| url   | The URL of the resource including the Wasm binary to instantiate. The supported schemes include "file://". The path of a "file://" URL is relative to the Dapr process unless it begins with '/'. | true     | "file://hello.wasm" |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
| url   | The URL of the resource including the Wasm binary to instantiate. The supported schemes include `file://`. The path of a `file://` URL is relative to the Dapr process unless it begins with `/`. | true     | `file://hello.wasm` |
```

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 5]
```diff
@@ -109,7 +110,7 @@ func handleRequest(req api.Request, resp api.Response) (next bool, reqCtx uint32
 ```
 
 If using TinyGo, compile as shown below and set the spec metadata field named
-"path" to the location of the output (ex "router.wasm"):
+"url" to the location of the output (ex "file://router.wasm"):
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
"url" to the location of the output (ex "file://router.wasm"):
```
```suggestion
"url" to the location of the output (for example, `file://router.wasm`):
```

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 6]
```diff
@@ -145,7 +146,7 @@ func rewrite(requestURI []byte) ([]byte, error) {
 ```
 
 If using TinyGo, compile as shown below and set the spec metadata field named
-"path" to the location of the output (ex "example.wasm"):
+"url" to the location of the output (ex "file://example.wasm"):
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
"url" to the location of the output (for example, `file://example.wasm`):
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick review - thanks!

**Comment by codefromthecrypt** [comment]
Thanks again @hhunter-ms!

**Comment by msfussell** [comment]
@codefromthecrypt  - You will need to perform a new pull on Dapr v1.11 branch for the build to succeed. Thx

**Review by hhunter-ms — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks again @hhunter-ms and @msfussell!

**Pull request merged**